internal/provider/data_source_dirhash_sha256.go

```diff
@@ -0,0 +1,60 @@
+package provider
+
+import (
+	"context"
+
+	"github.com/Think-iT-Labs/dirhash/lib"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataSourceDirhashSha256() *schema.Resource {
+	return &schema.Resource{
+		Description: "Calculate SHA256 checksum of a given directory",
+
+		ReadContext: dataSourceDirhashSha256Read,
+
+		Schema: map[string]*schema.Schema{
+			"directory": {
+				Type:        schema.TypeString,
+				Description: "Directory to hash.",
+				Required:    true,
+			},
+			"ignore": {
+				Type:        schema.TypeList,
+				Description: "List of ignored glob patterns.",
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Optional:    true,
+			},
+			"checksum": {
+				Type:        schema.TypeString,
+				Description: "Directory SHA 256 Checksum",
+				Computed:    true,
+			},
+		},
+	}
+}
+
+func dataSourceDirhashSha256Read(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
```

Comment:
```suggestion
func dataSourceDirhashSha256Read(ctx context.Context, rd *schema.ResourceData, meta any) diag.Diagnostics {
```
I suggest the second argument be named `rd` for ResourceData. This also minimizes confusion as `d` can refer to directory